funcx_endpoint/funcx_endpoint/executors/high_throughput/interchange.py

```diff
@@ -590,15 +591,22 @@ def _status_report_loop(self, kill_event, status_report_queue: queue.Queue):
         log.debug("Status reporting loop starting")
         log.info(f"Endpoint id: {self.endpoint_id}")
 
+        last = time.time() - self.heartbeat_period
+
         while not kill_event.is_set():
-            log.trace(f"Endpoint id : {self.endpoint_id}, {type(self.endpoint_id)}")
-            msg = EPStatusReport(
-                self.endpoint_id, self.get_status_report(), self.task_status_deltas
-            )
-            log.debug("Sending status report to executor, and clearing task deltas.")
-            status_report_queue.put(msg.pack())
-            self.task_status_deltas.clear()
-            time.sleep(self.heartbeat_period)
+            if last + self.heartbeat_period < time.time():
+                log.trace(f"Endpoint id : {self.endpoint_id}, {type(self.endpoint_id)}")
+                msg = EPStatusReport(
+                    self.endpoint_id, self.get_status_report(), self.task_status_deltas
+                )
+                log.debug(
+                    "Sending status report to executor, and clearing task deltas."
+                )
+                status_report_queue.put(msg.pack())
+                self.task_status_deltas.clear()
+
+                last = time.time()
+            time.sleep(0.1)
```

Comment:
The key point this change tries to address is ability to "instantly" move when `kill_event` is set, yes?  In that case, the only problem with the original hunk is use of `time.sleep`.  Consider `wait`, instead:
```python
while not kill_event.is_set():
    ... # loop kernel
    kill_event.wait(self.heartbeat_period)
```
which is now _actually instant_ when the event is set.

Reply:
Yes, good idea